tasks/parse-pdr/index.js

```diff
@@ -36,6 +37,17 @@ function parsePdr(event) {
   return parse.ingest()
     .then((payload) => {
       if (parse.connected) parse.end();
+      // opportunity to filter the granules of interest based on regex in granuleIdExtraction
+      if (config && config.collection && config.collection.granuleIdExtraction) {
+        const granuleIdExtraction = config.collection.granuleIdExtraction;
+        const granules = payload.granules.filter((g) => g.files[0].name.match(granuleIdExtraction));
+        const filteredPayload = {
+          granules: granules,
```

Comment:
This is updating the granules and granulesCount fields, but it is leaving the filesCount and totalSize fields with their original values (which will not probably be wrong).

Reply:
That's what I thought.  Leaving it as is, right?

Reply:
Just realized I had a typo in my comment.  Instead of "which will not probably be wrong", it should have said " which will _now_ probably be wrong."

This is removing granules from the return value, but is not adjusting the filesCount or totalSize fields, which will still reflect the values from before the granules were filtered out.